File: src/FileSling.Web/typescript/Model/DirectoryMetadata.ts
interface DirectoryMetadata {
    DirectoryId: string;

    CreatedAt: Date;
    ExpiresAt?: Date | null;
    LastFileUploadAt?: Date | null;

    MaxStorageBytes: number;
    UsedStorageBytes: number;

    // Protected properties
    DisplayName: string;
    ChallengePassword: string;
}

interface FileMetadata {
    DirectoryId: string;
    FileId: string;

    CreatedAt: Date;
    DownloadCount: number;
    FileSizeBytes: number;

    // Protected properties
    FileName: string;
    MimeType: string;
}

File: src/FileSling.Web/Model/DirectoryMetadata.cs
namespace Th11s.FileSling.Web.Model;

public record DirectoryMetadata(
    string DirectoryId,
    DateTimeOffset CreatedAt,
    DateTimeOffset? ExpiresAt,
    DateTimeOffset? LastFileUploadAt,

    long MaxStorageBytes,
    long UsedStorageBytes,

    byte[] Protected,
    byte[] ProtectedChallengePassword
);
